app/services/process_stripe_payment.rb

```diff
@@ -0,0 +1,25 @@
+class ProcessStripePayment
+  attr_reader :controller, :amount, :email, :token
+  
+  def initialize(controller, amount, email, token)
```

Comment:
Yea I could tell you but it is a week 7 stuff and you better hear it from Kevin :) In general you are working with a service object which can have an error_message attribute, so what you can do is to return that object back to controller and user instance method in unhappy path to retrieve the error message. Definitely good job thinking about abstracting it to service object.